x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/routes/upload_csv.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Logger, StartServicesAccessor } from '@kbn/core/server';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes/utils';
+import { schema } from '@kbn/config-schema';
+import Papa from 'papaparse';
+import type { StartPlugins } from '../../../../plugin';
+import type { AssetCriticalityCsvUploadResponse } from '../../../../../common/api/entity_analytics';
+import { CRITICALITY_CSV_MAX_SIZE_BYTES_WITH_TOLERANCE } from '../../../../../common/entity_analytics/asset_criticality';
+import type { ConfigType } from '../../../../config';
+import type { HapiReadableStream, SecuritySolutionPluginRouter } from '../../../../types';
+import { ASSET_CRITICALITY_CSV_UPLOAD_URL, APP_ID } from '../../../../../common/constants';
+import { checkAndInitAssetCriticalityResources } from '../check_and_init_asset_criticality_resources';
+import { transformCSVToUpsertRecords } from '../transform_csv_to_upsert_records';
+import { createAssetCriticalityProcessedFileEvent } from '../../../telemetry/event_based/events';
+
+export const assetCriticalityCSVUploadRoute = (
+  router: SecuritySolutionPluginRouter,
+  logger: Logger,
+  config: ConfigType,
+  getStartServices: StartServicesAccessor<StartPlugins>
+) => {
+  const { batchSize } = config.entityAnalytics.assetCriticality.csvUpload;
+  router.versioned
+    .post({
+      access: 'internal',
+      path: ASSET_CRITICALITY_CSV_UPLOAD_URL,
+      options: {
+        tags: ['access:securitySolution', `access:${APP_ID}-entity-analytics`],
+        body: {
+          output: 'stream',
+          accepts: 'multipart/form-data',
+          maxBytes: CRITICALITY_CSV_MAX_SIZE_BYTES_WITH_TOLERANCE,
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: schema.object({
+              file: schema.stream(),
+            }),
+          },
+        },
+      },
+      async (context, request, response) => {
+        const start = new Date();
+        const siemResponse = buildSiemResponse(response);
+
+        try {
+          await checkAndInitAssetCriticalityResources(context, logger);
```

Comment:
Right here, we'll want to `assertAdvancedSettingsEnabled`, as seen in other asset criticality routes such as [this one](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/routes/get.ts#L41)